lib/components/viewers/stop-viewer.js

```diff
@@ -423,19 +424,30 @@ function getFormattedStopTime (stopTime, homeTimezone, soonText = 'Due', timeFor
 
   const now = moment().tz(homeTimezone)
   const serviceDay = moment(stopTime.serviceDay * 1000).tz(homeTimezone)
+  // Determine if arrival occurs on different day, making sure to account for
+  // any extra days added to the service day if it arrives after midnight. Note:
+  // this can handle the rare (and non-existent?) case where an arrival occurs
+  // 48:00 hours (or more) from the start of the service day.
+  const departureTimeRemainder = stopTime.realtimeDeparture % ONE_DAY_IN_SECONDS
+  const daysAfterServiceDay = (stopTime.realtimeDeparture - departureTimeRemainder) / ONE_DAY_IN_SECONDS
+  const arrivalDay = serviceDay.add(daysAfterServiceDay, 'day')
+  // Calculate current home seconds after midnight.
   const currentHomeSecondsAfterMidnight = getHomeSecondsAfterMidnight(homeTimezone)
-  const differentDay = now.date() !== serviceDay.date()
-
-  // Determine whether to show departure as countdown (e.g. "5 min") or as HH:mm time
-  const secondsUntilDeparture = stopTime.realtimeDeparture - currentHomeSecondsAfterMidnight
-  const departsInFuture = stopTime.realtimeDeparture > currentHomeSecondsAfterMidnight
-  // Show the exact time if the departure occurs after midnight and if the
-  // departure happens within an hour.
-  // FIXME: It's unclear why this was designed to show exact time after midnight.
-  //  We should determine whether this is the desired behavior.
-  const showCountdown = !differentDay &&
-    secondsUntilDeparture < ONE_HOUR_IN_SECONDS &&
-    departsInFuture
+  const differentDay = now.dayOfYear() !== arrivalDay.dayOfYear()
+  // Determine whether to show departure as countdown (e.g. "5 min") or as HH:mm
+  // time. If arrival occurs on different (AKA future) day, use unaltered stop
+  // time departure. If the arrival occurs on the same day, use remainder
+  // departure time so that we're comparing times based on actual seconds since
+  // midnight (without interference from times that occur greater than 24 hours
+  // after start of service).
+  const secondsUntilDeparture = differentDay
+    ? stopTime.realtimeDeparture - currentHomeSecondsAfterMidnight
```

Comment:
I don't think this will work properly for the edge case of a departure time occurring more than 48 hours from the beginning service day. This would be a great opportunity to make a test case for this specific edge case.

Reply:
Can you clarify why you don't think this will work? Let's say it is Wednesday at 5pm (`61200 `) and we have a stop time with a Wednesday service day and a departure time at `172860` (12:01am Friday morning). `secondsUntilDeparture` will be `172860 - 61200 = 111660` or ~ 31 hours. From what I can tell this is the answer we're looking for.

As far as a test case for this -- I don't necessarily agree that it is worth our time to write a test case for an extreme edge case like this.

Reply:
Ok, try Thursday at 5pm (`61200` for today, but actually `147600` from the service day before). This will still output `111660` or ~ 31 hours when it should be just `25260` or ~ 7 hours.

Also, in my example feed below, I think this shows that departure times occurring after 48:00:00 are not extreme edge cases. It's quite common actually in Portland where there is a train route that take more than 2 days to get there from Chicago.